scripts/sheets_sync.py

```diff
@@ -270,48 +311,61 @@ def hq_updates(sheet_dict: dict, hq, sheet: str, hq_worksheet, hub: dict):
-            now = datetime.datetime.now(timezone.utc)
-            now_str = datetime.datetime.strftime(now, '%m/%d/%Y %H:%M:%S')
-            sheet_data_append = (
-                [
-                    sheet_dict[i][1:signup_columns['Zipcode']] + ['HOT LEAD'] +[now_str] + ['' for x in range(7)] +
-                    [sheet_dict[i][concat_col_idx]] +[sheet_dict[i][signup_columns['Zipcode']]]
-                        for i in sheet_dict
-                ]
-                                )
-
-            sheet_data_append.insert(0, hq_columns_list)
-            
+            # Update concatenated data entry field for existing records in hq sheet
+            range3 = hq_column_letters['data_entry_data'] + '4:' + hq_column_letters['data_entry_data']
+            hq_worksheet.update(range3, updates)
+            # Update in unrestricted sheet
+            range4 = (unrestricted_column_letters['data_entry_data'] + '4:' +
+                      unrestricted_column_letters['data_entry_data'])
+            unrestricted_sheet.update(range4, updates)
         except HttpError as e:
             log_error(e, 'sheets_sync', 'Error while updating data entry data column', hq_errors, hub)
 
-    sheet_parsons_append = Table(sheet_data_append)
-    return sheet_parsons_append
+    # Convert remainder of sheet_dict rows to a parsons table (to append to the bottom of hq)
+    #First turn the dict into an array
+    sheet_dict_array = [sheet_dict[email] for email in sheet_dict]
+    # Create a list for the col headers and insert into the array
+    if sheet == 'form responses':
+        sheet_dict_cols = list(signup_columns.keys()) + ['interest_form_responses']
+    elif sheet == 'data entry sheet':
+        sheet_dict_cols = list(signup_columns.keys()) + ['data_entry_data']
+    sheet_dict_array.insert(0,sheet_dict_cols)
+    #convert to table
+    sheet_dict_table = Table(sheet_dict_array)
+    # Since there is no timestamp field for the data entry sheet, assign today's date as the date_joined value
+    now = datetime.datetime.now(timezone.utc)
+    now_str = datetime.datetime.strftime(now, '%m/%d/%Y %H:%M:%S')
+    sheet_dict_table.fillna_column('date_joined',now_str)
+
+    # Create empty tbl with HQ columns in the order they appear in the spreadhseet and then fill the table
+    # with the new values from sheet dict. Really all we're doing here is ensuring that the new rows
+    # we're appending in HQ are in the correct order
+    hq_append_tbl = Table([hq_columns_list])
+    hq_append_tbl.concat(sheet_dict_table)
+    hq_append_tbl.fillna_column('status','HOT LEAD')
+    hq_append_tbl.move_column('status',hq_columns['status'])
+    if sheet == 'form responses':
+        hq_append_tbl.fillna_column('source', 'Interest Form')
+    if sheet == 'data entry sheet':
+        hq_append_tbl.fillna_column('source', 'Data Entry Sheet')
+    hq_append_tbl.move_column('source', hq_columns['source'])
+
```

Comment:
The next several lines prepare a table to append to the unrestricted sheet, and then the function returns two table instead of one. The first table is appended to hq, the second to the unrestricted sheet